### PR TITLE
Fix for make_hgrid segmentation faults

### PR DESCRIPTION
**Pull request by ceblanton**

### CODE REVIEW

File: tools/make_hgrid/make_hgrid.c [hunk 1]
```diff
@@ -842,107 +840,108 @@ int main(int argc, char* argv[])
...
     }
 
     //if(nest_grids >= 1) {
     for (n=0; n < nest_grids; n++) {
 
       if(refine_ratio[n] == 0) mpp_error("make_hgrid: --refine_ratio must be set when --nest_grids is set");
       if(parent_tile[n] == 0 && mpp_pe()==mpp_root_pe()) {
-	fprintf(stderr,"NOTE from make_hgrid: parent_tile is 0, the output grid will have resolution refine_ration*nlon");
+        fprintf(stderr,"NOTE from make_hgrid: parent_tile is 0, the output grid will have resolution refine_ration*nlon\n");
+        //ntiles++;//Only the six faces may be needed for output, but this is to prevent invalid mem access //MZ2
```
Comment (ceblanton) [inline tools/make_hgrid/make_hgrid.c:935]: Is this a debugging comment line that should be removed?

Comment (ngs333) [reply tools/make_hgrid/make_hgrid.c:935]: Ooops! Will remove.

File: tools/make_hgrid/create_gnomonic_cubic_grid.c [hunk 2]
```diff
@@ -166,33 +168,36 @@ void create_gnomonic_cubic_grid( char* grid_type, int *nlon, int *nlat, double *
...
 
   ntiles2=ntiles;
   global_nest=0;
 
-  if(num_nest_grids && parent_tile[0]== 0)
+  if(num_nest_grids && parent_tile[0]== 0){
     global_nest = 1;
+    //TODO: IMA related. Nncrease ntiles2 by 1 and will decrease some IMA below, but will also change to
+    //  0 two angle_dx and angle_dy values from GR pre multi-nest on tile1.nc
+  }
```
Comment (ceblanton) [inline tools/make_hgrid/create_gnomonic_cubic_grid.c:183]: Was this TODO intended for longer-term or for this issue? I think technical comments like this are very helpful, just wanted to check. Maybe define IMA?

Comment (ceblanton) [reply tools/make_hgrid/create_gnomonic_cubic_grid.c:183]: (Whoops, I see you defined IMA a few lines below!)

File: tools/make_hgrid/create_gnomonic_cubic_grid.c [hunk 3]
```diff
@@ -248,15 +255,17 @@ void create_gnomonic_cubic_grid( char* grid_type, int *nlon, int *nlat, double *
   }
 
   /* Use of these arrays permits different sized nests */
-  for (nn=0; nn < num_nest_grids; nn++) {
-			 nxl[nn+ntiles] = nx_nest_arr[nn];
-    nyl[nn+ntiles] = ny_nest_arr[nn];
-    nil[nn+ntiles] = ni_nest_arr[nn];
-    njl[nn+ntiles] = nj_nest_arr[nn];
+  if (global_nest != 1) {//DONE: For removing IMA and a core dump with GR.
```
Comment (ceblanton) [inline tools/make_hgrid/create_gnomonic_cubic_grid.c:258]: Same question/check. Is this a testing artifact?

File: tools/make_hgrid/create_gnomonic_cubic_grid.c [hunk 4]
```diff
@@ -654,22 +672,24 @@ void create_gnomonic_cubic_grid( char* grid_type, int *nlon, int *nlat, double *
...
   if (output_length_angle) {
     /*calculate rotation angle, just some workaround, will modify this in the future. */
     calc_rotation_angle2(nxp, x, y, angle_dx, angle_dy );
 
-    /* since angle is used in the model, set angle to 0 for nested region */
+    //since angle is used in the model, set angle to 0 for nested region
     for(nn=0; nn < num_nest_grids; nn++) {
-      for(i=0; i<=(nx_nest_arr[nn]+1)*(ny_nest_arr[nn]+1); i++) {
+      //TODO: IMA (Changing "<=" to < reduces one IMA but changes 4 numbers from GR pre-nested):
+      for(i=0; i<(nx_nest_arr[nn]+1)*(ny_nest_arr[nn]+1); i++) {
```
Comment (ceblanton) [inline tools/make_hgrid/create_gnomonic_cubic_grid.c:692]: This TODO is done!

File: tools/make_hgrid/make_hgrid.c [hunk 5]
```diff
@@ -1040,11 +1045,11 @@ int main(int argc, char* argv[])
...
     size3 = (unsigned long) (nxp+1) * nyp     * ntiles_global;
     size4 = (unsigned long) nxp     * nyp     * ntiles_global;
-  
-    //    if( nest_grids) { /* nest grid is the last tile */
+
+    //    if( nest_grids) {//MZ3 /* nest grid is the last tile */
     for (n_nest = ntiles_global; n_nest < ntiles_global + nest_grids; n_nest++) { /* nest grid is the last tile */
       if (verbose) fprintf(stderr, "[INFO] Adding memory size for nest %d, nest_grids: %d\n", n_nest, nest_grids);
-      size1 += (nxl[n_nest]+1) * (nyl[n_nest]+1);
+      size1 += (nxl[n_nest]+1) * (nyl[n_nest]+1);//MZ3
```
Comment (ceblanton) [inline tools/make_hgrid/make_hgrid.c:1052]: A couple harmless MZ3 debugging text strings :)

**Pull request closed without merging**